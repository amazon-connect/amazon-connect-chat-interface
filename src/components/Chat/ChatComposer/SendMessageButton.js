import React from 'react';
import styled from 'styled-components';
import defaultTheme from '../../../theme/defaultTheme';

const ACTIVE_COLOR = defaultTheme.palette.secondaryBlack;
const INACTIVE_COLOR = defaultTheme.palette.whisper;

const SendButton = styled.div`
  cursor: ${props => props.isActive ? 'pointer' : 'default'};

  &>svg {
    fill: ${props => props.isActive ? ACTIVE_COLOR : INACTIVE_COLOR};
  }
`;

/**
 * Send message button for the Chat Composer.
 * 
 * @param {Object} props
 * @param {boolean} props.isActive
 * @param {Function} props.sendMessage
 */
function SendMessageButton({ isActive, sendMessage }) {
  return (
    <SendButton isActive={isActive} onClick={sendMessage} data-testid={'customer-chat-send-message-button'} aria-label={'Send Message'}>
      <svg xmlns="http://www.w3.org/2000/svg" height="24px" viewBox="0 0 24 24" width="24px"><path d="M0 0h24v24H0z" fill="none"/><path d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"/></svg>
    </SendButton>
  );
};

export default SendMessageButton;