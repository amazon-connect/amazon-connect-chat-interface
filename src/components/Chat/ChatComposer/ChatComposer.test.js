// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: MIT-0

import React from "react";
import { IntlProvider } from 'react-intl';
import ChatComposer from "./ChatComposer";
import { ThemeProvider } from "../../../theme";
import { render, fireEvent, screen, prettyDOM } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { ContentType } from "../datamodel/Model";
import { KEYBOARD_KEY_CONSTANTS } from "connect-constants";
import "@testing-library/jest-dom";

const mockAttachmentsFile = {
  name: "testUpload.pdf",
  type: ContentType.ATTACHMENT_CONTENT_TYPE.PDF,
  size: 1,
};

let mockComposer;
let mockProps;

function renderElement(props) {
  mockComposer = render(
    <ThemeProvider>
      <IntlProvider
          locale="en"
          key="en"
          messages={{}}>
        <ChatComposer {...props}/>
      </IntlProvider>
    </ThemeProvider>
  );
}

jest.useFakeTimers();
jest.spyOn(global, "setTimeout");
describe("when window.connect is not defined", () => {
  beforeEach(() => {
    const onTyping = jest.fn().mockResolvedValue(undefined);
    const addMessage = jest.fn().mockResolvedValue(undefined);
    const addAttachment = jest.fn().mockResolvedValue(undefined);
    mockProps = {
      onTyping: onTyping,
      addAttachment: addAttachment,
      addMessage: addMessage,
      contactId: "12344",
      contactStatus: "connected",
      typedMessage: "",
      composerConfig: { attachmentsEnabled: true },
    };
    navigator.__defineGetter__("userAgent", function () {
      return "Mozilla/5.0 (iPhone; CPU iPhone OS 15_5 like Mac OS X)";
    });
  });

  test("Style should match the snapshot", () => {
    renderElement(mockProps);
    expect(mockComposer).toMatchSnapshot();
  });

  test("Should not be able to see the paperclip icon without permission", () => {
    mockProps.composerConfig.attachmentsEnabled = false;
    renderElement(mockProps);
    expect(mockComposer.queryByTestId("customer-chat-file-select")).toBeNull();
  });

  test("Should be able to send an attachment", () => {
    renderElement(mockProps);
    const fileInput = mockComposer.getByTestId("customer-chat-file-select");
    fireEvent.change(fileInput, { target: { files: [mockAttachmentsFile] } });
    const textInput = mockComposer.getByTestId("customer-chat-text-input");
    fireEvent.keyDown(textInput, { key: KEYBOARD_KEY_CONSTANTS.ENTER });
    expect(mockProps.addAttachment).toHaveBeenCalledTimes(1);
    expect(mockProps.addAttachment).toHaveBeenCalledWith(mockProps.contactId, {
      ...mockAttachmentsFile,
    });
  });

  test("Should be able to send an attachment via the send message button", () => {
    renderElement(mockProps);

    const fileInput = mockComposer.getByTestId("customer-chat-file-select");
    fireEvent.change(fileInput, { target: { files: [mockAttachmentsFile] } });

    const sendMessageButton = mockComposer.getByTestId("customer-chat-send-message-button");
    fireEvent.click(sendMessageButton);

    expect(mockProps.addAttachment).toHaveBeenCalledTimes(1);
    expect(mockProps.addAttachment).toHaveBeenCalledWith(mockProps.contactId, {
      ...mockAttachmentsFile,
    });
  });

  test("Should be able to send a message via the send message button", async () => {
    renderElement(mockProps);

    const testMessage = "Hello, World!";
    const textInput = mockComposer.getByTestId("customer-chat-text-input");
    userEvent.type(textInput, testMessage);

    const sendMessageButton = mockComposer.getByTestId("customer-chat-send-message-button");
    fireEvent.click(sendMessageButton);

    expect(mockProps.addMessage).toHaveBeenCalledTimes(1);
    expect(mockProps.addMessage).toHaveBeenCalledWith(mockProps.contactId, {
      text: testMessage,
    });
  });

  test("Should be able to jitter to fix iphone mobile scroll issue", () => {
    renderElement(mockProps);

    const testMessage = "Hello, World!";
    const textInput = mockComposer.getByTestId("customer-chat-text-input");
    userEvent.type(textInput, testMessage);
    expect(setTimeout).toHaveBeenCalledWith(expect.any(Function), 300);
    jest.runOnlyPendingTimers();
  });

  test("Should be able to send an attachment plus additional text via the send message button", () => {
    renderElement(mockProps);

    const fileInput = mockComposer.getByTestId("customer-chat-file-select");
    fireEvent.change(fileInput, { target: { files: [mockAttachmentsFile] } });

    const testMessage = "Hello, World!";
    const textInput = mockComposer.getByTestId("customer-chat-text-input");
    userEvent.type(textInput, testMessage);

    const sendMessageButton = mockComposer.getByTestId("customer-chat-send-message-button");
    fireEvent.click(sendMessageButton);

    expect(mockProps.addAttachment).toHaveBeenCalledTimes(1);
    expect(mockProps.addAttachment).toHaveBeenCalledWith(mockProps.contactId, {
      ...mockAttachmentsFile,
    });

    expect(mockProps.addMessage).toHaveBeenCalledTimes(1);
    expect(mockProps.addMessage).toHaveBeenCalledWith(mockProps.contactId, {
      text: testMessage,
    });
  });

  test("Should not send message if send button is clicked and there is no input", () => {
    renderElement(mockProps);
    const sendMessageButton = mockComposer.getByTestId("customer-chat-send-message-button");
    fireEvent.click(sendMessageButton);

    expect(mockProps.addMessage).toHaveBeenCalledTimes(0);
  });

  test("Should be able to unselect an attachment", () => {
    renderElement(mockProps);
    const fileInput = mockComposer.getByTestId("customer-chat-file-select");
    fireEvent.change(fileInput, { target: { files: [mockAttachmentsFile] } });
    const textInput = mockComposer.getByTestId("customer-chat-text-input");
    fireEvent.keyDown(textInput, { key: KEYBOARD_KEY_CONSTANTS.DELETE });
    fireEvent.keyDown(textInput, { key: KEYBOARD_KEY_CONSTANTS.ENTER });
    expect(mockProps.addAttachment).toHaveBeenCalledTimes(0);
  });

  test("Should be able to click send button using Tab and Space", () => {
    renderElement(mockProps);
    const textInput = document.querySelector('[aria-label="Type a message"]');
    const testMessage = "Hello, World!";
    userEvent.type(textInput, testMessage);
    userEvent.tab();
    const sendMessageButton = mockComposer.getByTestId("customer-chat-send-message-button");
    expect(sendMessageButton).toHaveFocus();
    fireEvent.keyDown(sendMessageButton, { key: KEYBOARD_KEY_CONSTANTS.SPACE });
    expect(mockProps.addMessage).toHaveBeenCalledTimes(1);
  });

  test("Should be able to click send button using Tab and Enter", () => {
    renderElement(mockProps);
    const textInput = document.querySelector('[aria-label="Type a message"]');
    const testMessage = "Hello, World!";
    userEvent.type(textInput, testMessage);
    userEvent.tab();
    const sendMessageButton = mockComposer.getByTestId("customer-chat-send-message-button");
    expect(sendMessageButton).toHaveFocus();
    fireEvent.keyDown(sendMessageButton, { key: KEYBOARD_KEY_CONSTANTS.ENTER });
    expect(mockProps.addMessage).toHaveBeenCalledTimes(1);
  });

  test("Should be able to click attachment icon using Tab and Space", () => {
    renderElement(mockProps);
    const textInput = document.querySelector('[aria-label="Type a message"]');
    const testMessage = "Hello, World!";
    userEvent.type(textInput, testMessage);
    // focus on the attachment icon
    userEvent.tab({ shift: true });
    expect(document.querySelector('[aria-label="Attach a file"]')).toHaveFocus();
    const attachmentIcon = mockComposer.getByTestId("customer-chat-attachment-icon");
    // TODO: add test for verifying the click event
    fireEvent.keyDown(attachmentIcon, { key: KEYBOARD_KEY_CONSTANTS.SPACE });
  });

  test("Should be able to click attachment icon using Tab and Enter", () => {
    renderElement(mockProps);
    const textInput = document.querySelector('[aria-label="Type a message"]');
    const testMessage = "Hello, World!";
    userEvent.type(textInput, testMessage);
    // focus on the attachment icon
    userEvent.tab({ shift: true });
    expect(document.querySelector('[aria-label="Attach a file"]')).toHaveFocus();
    const attachmentIcon = mockComposer.getByTestId("customer-chat-attachment-icon");
    // TODO: add test for verifying the click event
    fireEvent.keyDown(attachmentIcon, { key: KEYBOARD_KEY_CONSTANTS.ENTER });
  });
});

describe("when window.connect is defined", () => {
  beforeEach(() => {
    window.connect = {
      LogManager: {
        getLogger: function (obj) {
          return {
            debug: jest.fn(),
            info: jest.fn(),
            error: jest.fn(),
          };
        },
      },
    };
  });

  test("Style should match the snapshot", () => {
    renderElement(mockProps);
    expect(mockComposer).toMatchSnapshot();
  });

  test("Should be able to send an attachment via the send message button", () => {
    renderElement(mockProps);

    const fileInput = mockComposer.getByTestId("customer-chat-file-select");
    fireEvent.change(fileInput, { target: { files: [mockAttachmentsFile] } });

    const sendMessageButton = mockComposer.getByTestId("customer-chat-send-message-button");
    fireEvent.click(sendMessageButton);

    expect(mockProps.addAttachment).toHaveBeenCalledTimes(1);
    expect(mockProps.addAttachment).toHaveBeenCalledWith(mockProps.contactId, {
      ...mockAttachmentsFile,
    });
  });

  it("should not render richtoolbar when missing optional supportedMessagingContentTypes input", () => {
    mockProps.composerConfig.richMessagingEnabled = false;
    delete mockProps.composerConfig.supportedMessagingContentTypes;
    renderElement(mockProps);

    expect(() => mockComposer.getByTestId("rich-text-editor")).toThrow("Unable to find an element");
  });

  it("should render richtoolbar when given supportedMessagingContentTypes input", () => {
    mockProps.composerConfig.richMessagingEnabled = true;
    mockProps.composerConfig.supportedMessagingContentTypes = "text/plain,text/markdown";
    renderElement(mockProps);

    const richTextEditor = mockComposer.getByTestId("rich-text-editor");
    expect(richTextEditor).toBeInTheDocument();
  });
});
