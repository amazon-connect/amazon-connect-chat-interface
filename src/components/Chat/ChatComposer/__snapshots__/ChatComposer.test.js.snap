// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when window.connect is defined Style should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    .c7 {
  cursor: default;
}

.c7>svg {
  fill: #EDEDED;
}

.c0 {
  margin: 0;
  padding: 0;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #FFFFFF;
  border: 0.5px solid #CCCCCC;
  border-left: 0;
  border-right: 0;
}

.c6 {
  position: absolute;
  padding: 10px;
  padding-right: 14px;
  top: 0;
  right: 0;
  z-index: 2;
}

.c2 {
  cursor: pointer;
  height: auto;
  vertical-align: top;
}

.c2 button {
  height: 100%;
  width: 100%;
}

.c2 label {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  font-size: 0;
  height: 100%;
  padding-left: 6px;
  padding-right: 6px;
  margin-bottom: 0;
}

.c2 input {
  display: none;
}

.c2 + [role="textbox"] {
  padding-left: 0;
}

.c3 {
  background-color: transparent;
  border: 1px solid transparent;
  position: relative;
  padding: 0;
  margin: 0;
}

.c5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  outline: none;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
  word-break: break-word;
  font-family: inherit;
  font-size: 1rem !important;
  padding: 10px;
  padding-left: 0;
  padding-right: 44px;
  margin-left: 10px;
  max-height: 80px;
  line-height: 1.5rem;
  overflow: auto;
  min-height: 39px;
  z-index: 2;
  resize: none;
  font-size: 0.875rem;
  border: none;
}

.c5::-webkit-input-placeholder {
  color: #666666;
}

.c5::-moz-placeholder {
  color: #666666;
}

.c5:-ms-input-placeholder {
  color: #666666;
}

.c5::placeholder {
  color: #666666;
}

.c5:focus::-webkit-input-placeholder {
  color: transparent;
}

.c5:focus::-moz-placeholder {
  color: transparent;
}

.c5:focus:-ms-input-placeholder {
  color: transparent;
}

.c5:focus::placeholder {
  color: transparent;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 0;
}

.c4 svg {
  width: 24px;
  height: 24px;
}

<div>
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <div
            class="c2"
            data-testid="customer-chat-attachment-icon"
            tabindex="0"
          >
            <button
              aria-label="Attach a file"
              class="c3"
            >
              <label
                for="customer-chat-file-select-12344"
              >
                <div
                  class="c4"
                >
                  <svg
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M16.5 6v11.5c0 2.21-1.79 4-4 4s-4-1.79-4-4V5c0-1.38 1.12-2.5 2.5-2.5s2.5 1.12 2.5 2.5v10.5c0 .55-.45 1-1 1s-1-.45-1-1V6H10v9.5c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5V5c0-2.21-1.79-4-4-4S7 2.79 7 5v12.5c0 3.04 2.46 5.5 5.5 5.5s5.5-2.46 5.5-5.5V6h-1.5z"
                    />
                  </svg>
                </div>
                <input
                  accept="text/plain,application/pdf,image/jpeg,image/png,application/msword,application/vnd.openxmlformats-officedocument.wordprocessingml.document,application/vnd.ms-excel,application/vnd.openxmlformats-officedocument.spreadsheetml.sheet,text/csv,application/vnd.ms-powerpoint,application/vnd.openxmlformats-officedocument.presentationml.presentation,audio/wav,audio/x-wav,audio/vnd.wave,image/jpeg,application/rtf,application/x-rtf,image/heic,video/quicktime,video/mp4,text/richtext,text/rtf,.doc,.docx,.xls,.xlsx,.ppt,.pptx,.txt,.csv,.jfif,.rtf,.heic,.mov,.mp4"
                  aria-label="Attach a file"
                  data-testid="customer-chat-file-select"
                  id="customer-chat-file-select-12344"
                  tabindex="-1"
                  type="file"
                />
              </label>
            </button>
          </div>
          <textarea
            aria-label="Type a message"
            class="c5"
            data-testid="customer-chat-text-input"
            placeholder="Type a message"
            spellcheck="true"
            tabindex="0"
          />
          <div
            class="c6"
          >
            <div
              aria-label="Send Message"
              class="c7"
              data-testid="customer-chat-send-message-button"
              tabindex="0"
            >
              <svg
                height="24px"
                viewBox="0 0 24 24"
                width="24px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-kGXeez bsulgI"
    >
      <div
        class="sc-kpOJdX yKUIE"
      >
        <div
          class="sc-ckVGcZ bSJpNg"
          data-testid="customer-chat-attachment-icon"
          tabindex="0"
        >
          <button
            aria-label="Attach a file"
            class="sc-jKJlTe iesImO"
          >
            <label
              for="customer-chat-file-select-12344"
            >
              <div
                class="sc-kEYyzF gxVqSG"
              >
                <svg
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M16.5 6v11.5c0 2.21-1.79 4-4 4s-4-1.79-4-4V5c0-1.38 1.12-2.5 2.5-2.5s2.5 1.12 2.5 2.5v10.5c0 .55-.45 1-1 1s-1-.45-1-1V6H10v9.5c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5V5c0-2.21-1.79-4-4-4S7 2.79 7 5v12.5c0 3.04 2.46 5.5 5.5 5.5s5.5-2.46 5.5-5.5V6h-1.5z"
                  />
                </svg>
              </div>
              <input
                accept="text/plain,application/pdf,image/jpeg,image/png,application/msword,application/vnd.openxmlformats-officedocument.wordprocessingml.document,application/vnd.ms-excel,application/vnd.openxmlformats-officedocument.spreadsheetml.sheet,text/csv,application/vnd.ms-powerpoint,application/vnd.openxmlformats-officedocument.presentationml.presentation,audio/wav,audio/x-wav,audio/vnd.wave,image/jpeg,application/rtf,application/x-rtf,image/heic,video/quicktime,video/mp4,text/richtext,text/rtf,.doc,.docx,.xls,.xlsx,.ppt,.pptx,.txt,.csv,.jfif,.rtf,.heic,.mov,.mp4"
                aria-label="Attach a file"
                data-testid="customer-chat-file-select"
                id="customer-chat-file-select-12344"
                tabindex="-1"
                type="file"
              />
            </label>
          </button>
        </div>
        <textarea
          aria-label="Type a message"
          class="sc-hMqMXs hOdWwg"
          data-testid="customer-chat-text-input"
          placeholder="Type a message"
          spellcheck="true"
          tabindex="0"
        />
        <div
          class="sc-dxgOiQ jhaimo"
        >
          <div
            aria-label="Send Message"
            class="sc-bdVaJa iDYuWj"
            data-testid="customer-chat-send-message-button"
            tabindex="0"
          >
            <svg
              height="24px"
              viewBox="0 0 24 24"
              width="24px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`when window.connect is not defined Style should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c7 {
  cursor: default;
}

.c7>svg {
  fill: #EDEDED;
}

.c0 {
  margin: 0;
  padding: 0;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #FFFFFF;
  border: 0.5px solid #CCCCCC;
  border-left: 0;
  border-right: 0;
}

.c6 {
  position: absolute;
  padding: 10px;
  padding-right: 14px;
  top: 0;
  right: 0;
  z-index: 2;
}

.c2 {
  cursor: pointer;
  height: auto;
  vertical-align: top;
}

.c2 button {
  height: 100%;
  width: 100%;
}

.c2 label {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  font-size: 0;
  height: 100%;
  padding-left: 6px;
  padding-right: 6px;
  margin-bottom: 0;
}

.c2 input {
  display: none;
}

.c2 + [role="textbox"] {
  padding-left: 0;
}

.c3 {
  background-color: transparent;
  border: 1px solid transparent;
  position: relative;
  padding: 0;
  margin: 0;
}

.c5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  outline: none;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
  word-break: break-word;
  font-family: inherit;
  font-size: 1rem !important;
  padding: 10px;
  padding-left: 0;
  padding-right: 44px;
  margin-left: 10px;
  max-height: 80px;
  line-height: 1.5rem;
  overflow: auto;
  min-height: 39px;
  z-index: 2;
  resize: none;
  font-size: 0.875rem;
  border: none;
}

.c5::-webkit-input-placeholder {
  color: #666666;
}

.c5::-moz-placeholder {
  color: #666666;
}

.c5:-ms-input-placeholder {
  color: #666666;
}

.c5::placeholder {
  color: #666666;
}

.c5:focus::-webkit-input-placeholder {
  color: transparent;
}

.c5:focus::-moz-placeholder {
  color: transparent;
}

.c5:focus:-ms-input-placeholder {
  color: transparent;
}

.c5:focus::placeholder {
  color: transparent;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 0;
}

.c4 svg {
  width: 24px;
  height: 24px;
}

<body>
    <div>
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <div
            class="c2"
            data-testid="customer-chat-attachment-icon"
            tabindex="0"
          >
            <button
              aria-label="Attach a file"
              class="c3"
            >
              <label
                for="customer-chat-file-select-12344"
              >
                <div
                  class="c4"
                >
                  <svg
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M16.5 6v11.5c0 2.21-1.79 4-4 4s-4-1.79-4-4V5c0-1.38 1.12-2.5 2.5-2.5s2.5 1.12 2.5 2.5v10.5c0 .55-.45 1-1 1s-1-.45-1-1V6H10v9.5c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5V5c0-2.21-1.79-4-4-4S7 2.79 7 5v12.5c0 3.04 2.46 5.5 5.5 5.5s5.5-2.46 5.5-5.5V6h-1.5z"
                    />
                  </svg>
                </div>
                <input
                  accept="text/plain,application/pdf,image/jpeg,image/png,application/msword,application/vnd.openxmlformats-officedocument.wordprocessingml.document,application/vnd.ms-excel,application/vnd.openxmlformats-officedocument.spreadsheetml.sheet,text/csv,application/vnd.ms-powerpoint,application/vnd.openxmlformats-officedocument.presentationml.presentation,audio/wav,audio/x-wav,audio/vnd.wave,image/jpeg,application/rtf,application/x-rtf,image/heic,video/quicktime,video/mp4,text/richtext,text/rtf,.doc,.docx,.xls,.xlsx,.ppt,.pptx,.txt,.csv,.jfif,.rtf,.heic,.mov,.mp4"
                  aria-label="Attach a file"
                  data-testid="customer-chat-file-select"
                  id="customer-chat-file-select-12344"
                  tabindex="-1"
                  type="file"
                />
              </label>
            </button>
          </div>
          <textarea
            aria-label="Type a message"
            class="c5"
            data-testid="customer-chat-text-input"
            placeholder="Type a message"
            spellcheck="true"
            tabindex="0"
          />
          <div
            class="c6"
          >
            <div
              aria-label="Send Message"
              class="c7"
              data-testid="customer-chat-send-message-button"
              tabindex="0"
            >
              <svg
                height="24px"
                viewBox="0 0 24 24"
                width="24px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-kGXeez bsulgI"
    >
      <div
        class="sc-kpOJdX yKUIE"
      >
        <div
          class="sc-ckVGcZ bSJpNg"
          data-testid="customer-chat-attachment-icon"
          tabindex="0"
        >
          <button
            aria-label="Attach a file"
            class="sc-jKJlTe iesImO"
          >
            <label
              for="customer-chat-file-select-12344"
            >
              <div
                class="sc-kEYyzF gxVqSG"
              >
                <svg
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M16.5 6v11.5c0 2.21-1.79 4-4 4s-4-1.79-4-4V5c0-1.38 1.12-2.5 2.5-2.5s2.5 1.12 2.5 2.5v10.5c0 .55-.45 1-1 1s-1-.45-1-1V6H10v9.5c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5V5c0-2.21-1.79-4-4-4S7 2.79 7 5v12.5c0 3.04 2.46 5.5 5.5 5.5s5.5-2.46 5.5-5.5V6h-1.5z"
                  />
                </svg>
              </div>
              <input
                accept="text/plain,application/pdf,image/jpeg,image/png,application/msword,application/vnd.openxmlformats-officedocument.wordprocessingml.document,application/vnd.ms-excel,application/vnd.openxmlformats-officedocument.spreadsheetml.sheet,text/csv,application/vnd.ms-powerpoint,application/vnd.openxmlformats-officedocument.presentationml.presentation,audio/wav,audio/x-wav,audio/vnd.wave,image/jpeg,application/rtf,application/x-rtf,image/heic,video/quicktime,video/mp4,text/richtext,text/rtf,.doc,.docx,.xls,.xlsx,.ppt,.pptx,.txt,.csv,.jfif,.rtf,.heic,.mov,.mp4"
                aria-label="Attach a file"
                data-testid="customer-chat-file-select"
                id="customer-chat-file-select-12344"
                tabindex="-1"
                type="file"
              />
            </label>
          </button>
        </div>
        <textarea
          aria-label="Type a message"
          class="sc-hMqMXs hOdWwg"
          data-testid="customer-chat-text-input"
          placeholder="Type a message"
          spellcheck="true"
          tabindex="0"
        />
        <div
          class="sc-dxgOiQ jhaimo"
        >
          <div
            aria-label="Send Message"
            class="sc-bdVaJa iDYuWj"
            data-testid="customer-chat-send-message-button"
            tabindex="0"
          >
            <svg
              height="24px"
              viewBox="0 0 24 24"
              width="24px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
